plotting_functions.py

```diff
@@ -0,0 +1,1653 @@
+            ha="center",
+            va="center",
+            color="black",
+        )
+
+        ### Draw the horizontal lines at the mean values
+        axes[1].hlines(
+            y=mean[im_index],
+            xmin=-0.01,
+            xmax=std_ln[im_index],
+            color="black",
+            linestyle="--",
+        )
+
+        ### Draw the vertical lines at the standard deviation values
+        axes[1].vlines(
+            x=std_ln[im_index],
+            ymin=plot_ylims[0],
+            ymax=plot_ylims[1],
+            color="black",
+            linestyle="--",
+        )
+
+        ### Plot labels (A), (B), (C)
+        axes[1].text(
+            std_ln[im_index],
+            plot_ylims[1] + 0.5,
+            f"({annotation_labels[annotation_ims.index(annotation_im)]})",
+            ha="center",
+            va="center",
+            color="black",
+        )
+
+    axes[0].set_xlabel(r"Peak ground acceleration (g)")
+    axes[0].set_ylabel(r"Annual hazard probability, $\mu_{P(PGA=pga)}$")
+
+    axes[1].set_ylabel(r"Mean annual hazard probability, $\mu_{P(PGA=pga)}$")
+    axes[1].set_xlabel(r"Dispersion in hazard probability, $\sigma_{\ln P(PGA=pga)}$")
+
+    axes[0].text(
+        1.25e-4,
+        12,
+        "Dispersion in hazard probability",
+        ha="left",
+        va="center",
+        color="black",
+    )
+
+    text_row_2_height = plot_ylims[1] + 5.0
+
+    axes[0].text(
+        6e-3,
+        text_row_2_height,
+        "Reference point:",
+        ha="right",
+        va="center",
+        color="black",
+    )
+
+    axes[0].text(
+        annotation_ims[0],
+        text_row_2_height,
+        "(A)",
+        ha="center",
+        va="center",
+        color="black",
+    )
+
+    axes[0].text(
+        annotation_ims[1],
+        text_row_2_height,
+        "(B)",
+        ha="center",
+        va="center",
+        color="black",
+    )
+
+    axes[0].text(
+        annotation_ims[2],
+        text_row_2_height,
+        "(C)",
+        ha="center",
+        va="center",
+        color="black",
+    )
+
+    axes[0].text(
+        8e-3, plot_ylims[1] + 2.0, f"{im} =  ", ha="right", va="center", color="black"
+    )
+
+    axes[0].text(
+        5.8e-3,
+        plot_ylims[1] + 0.5,
+        rf"$\sigma_{{\ln P({im.upper()}={im.lower()})}} = $",
+        ha="right",
+        va="center",
+        color="black",
+    )
+
+    plt.subplots_adjust(bottom=0.1, top=0.81, left=0.085, right=0.99, wspace=0.23)
+
+    plt.savefig(
+        plot_output_directory
+        / f"{gmcm_name}_{location_short_name}_predictions_and_aggregate_stats.png",
+        dpi=plot_dpi,
+    )
+
+
+def make_figure_of_gmm_dispersion_ranges(
+    results_directory: Union[Path, str],
+    plot_output_directory: Union[Path, str],
+    locations: tuple[str, ...] = ("AKL", "WLG", "CHC"),
+    filter_strs: tuple[str, ...] = ("CRU", "HIK_and_PUY", "SLAB"),
```

Comment:
Flag?

Reply:
Flags could work for HIK_and_PUY but I would have had to change some of the condition checks so I've changed all these option values to StrEnum for the benefit of Enums but also a relatively simple implementation